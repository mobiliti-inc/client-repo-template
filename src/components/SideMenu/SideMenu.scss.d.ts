// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'fades-in-out': string;
  'side-menu': string;
  'visible': string;
}
declare var cssExports: CssExports;
export = cssExports;
