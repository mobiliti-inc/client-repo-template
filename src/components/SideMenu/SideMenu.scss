@import '../../styles/variables';

.side-menu {
    display: flex;
    background: linear-gradient(180deg, #FFFFFF 0%, rgba(255, 255, 255, 0) 0.01%, rgba(255, 255, 255, 0.25) 85.42%), linear-gradient(0deg, rgba(73, 150, 174, 0.8), rgba(73, 150, 174, 0.8)), radial-gradient(450.00px at 50% 50%, #FFFFFF 0%, rgba(255, 255, 255, 0) 100%);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    width: 10em;
    min-height: 100vh;
    color: $white;
    justify-content: center;
    padding: 24px 0;
    transition: transform .5s;


    @media (max-width: $mobile-width) {
        box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.25);
        transform:  translate(-1000px)
	}
}

.visible {
    transform:  translate(0)
}
