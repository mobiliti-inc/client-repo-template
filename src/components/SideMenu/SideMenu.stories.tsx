import { storiesOf } from "@storybook/react";
import { boolean, object } from "@storybook/addon-knobs";
import React from "react";
import { SideMenu, BrandLogo } from "../";

storiesOf("SideMenu", module)
	.add("No brand logo", () =>
		<SideMenu
			visible={boolean("visible", true)}
			style={ object('style', { })}/>
		)
	.add("With brand logo", () =>
		<SideMenu
			visible={boolean("visible", true)}
			style={ object('style', { })}>
			<BrandLogo />
		</SideMenu>);
