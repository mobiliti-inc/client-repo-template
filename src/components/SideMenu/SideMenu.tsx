import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

import './SideMenu.scss';

type customStyle = {
	background?: string;
	color?: string;
	height?: string;
	width?: string;
};

interface IProps {
	style?: customStyle;
	children?: ReactNode;
	visible?: Boolean;
}

const SideMenu: FC<IProps> = ({ style, children, visible }) => {
	return (
		<div style={style} styleName={classNames('side-menu', { visible } )}>
			{children}
		</div>
	);
};

export default SideMenu;
