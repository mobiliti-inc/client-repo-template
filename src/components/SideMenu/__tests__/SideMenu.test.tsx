import React from 'react';

import SideMenu from '../SideMenu';

describe('BrandLogo', () => {
	it('renders', () => {
		const subject = shallow(<SideMenu />);
		expect(subject).toMatchSnapshot();
	});

	it('renders with children', () => {
		const subject = shallow(<SideMenu>text</SideMenu>);
		expect(subject).toMatchSnapshot();
		const sideMenuDiv =subject.find('.side-menu');
		expect(sideMenuDiv).toHaveLength(1);
		expect(sideMenuDiv.props().children).toEqual('text');
	});

	it('renders with visible', () => {
		const subject = shallow(<SideMenu visible={true} />);
		expect(subject).toMatchSnapshot();

		const sideMenuDiv =subject.find('.visible');
		expect(sideMenuDiv).toHaveLength(1);
	});

	it('renders with style', () => {
		const customStyle = {'background': 'red'};
		const subject = shallow(<SideMenu style={customStyle} />);
		expect(subject).toMatchSnapshot();
		const sideMenuDiv =subject.find('.side-menu');
		expect(sideMenuDiv).toHaveLength(1);
		expect(sideMenuDiv.props().style.background).toEqual(customStyle.background);
	});
});
