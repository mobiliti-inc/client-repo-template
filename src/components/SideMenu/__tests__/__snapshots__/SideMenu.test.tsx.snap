// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo renders 1`] = `
<div
  className="side-menu"
/>
`;

exports[`BrandLogo renders with children 1`] = `
<div
  className="side-menu"
>
  text
</div>
`;

exports[`BrandLogo renders with style 1`] = `
<div
  className="side-menu"
  style={
    Object {
      "background": "red",
    }
  }
/>
`;

exports[`BrandLogo renders with visible 1`] = `
<div
  className="side-menu visible"
/>
`;
